include/bmi/AbstractCLibBmiAdapter.hpp

```diff
@@ -192,24 +97,18 @@ namespace models {
              * @throws ``::external::ExternalIntegrationException`` If symbol could not be found for the shared library.
              * @see dynamic_load_symbol(std::string, bool)
              */
-            inline void *dynamic_load_symbol(const std::string &symbol_name) {
-                return dynamic_load_symbol(symbol_name, false);
-            }
+            void *dynamic_load_symbol(const std::string &symbol_name);
 
-            inline const std::string &get_bmi_registration_function() {
-                return bmi_registration_function;
-            }
+            const std::string &get_bmi_registration_function();
 
-            inline const void *get_dyn_lib_handle() {
-                return dyn_lib_handle;
-            }
+            const void *get_dyn_lib_handle();
 
         private:
 
             /** Path to the BMI shared library file, for dynamic linking. */
             std::string bmi_lib_file;
             /** Name of the function that registers BMI struct's function pointers to the right module functions. */
-            const std::string bmi_registration_function;
+            std::string bmi_registration_function;
```

Comment:
Removed const specifier, since it isn't necessary here. Really, the getter for this should be const instead.

Reply:
Agree the getter could be marked const, but that would prevent future code added to the class from trying to change the `bmi_registration_function`, perhaps even inadvertently, which could lead to unexpected behavior.  I guess I'm trying to say, it doesn't hurt to be overly pedantic in these cases where we want the compiler to help with validation as much as possible.

Reply:
Reverted in [4f83059](https://github.com/NOAA-OWP/ngen/pull/750/commits/4f83059438c162f0cfe8fae81899a8c4e8891356)